src/mdio/converters/segy.py

```diff
@@ -291,16 +303,17 @@ def populate_non_dim_coordinates(
     return dataset, drop_vars_delayed
 
 
-def _get_horizontal_coordinate_unit(segy_headers: list[Dimension]) -> LengthUnitModel | None:
+def _get_horizontal_coordinate_unit(segy_info: SegyFileHeaderDump) -> LengthUnitModel | None:
     """Get the coordinate unit from the SEG-Y headers."""
-    name = TraceHeaderFieldsRev0.COORDINATE_UNIT.name.upper()
-    unit_hdr = next((c for c in segy_headers if c.name.upper() == name), None)
-    if unit_hdr is None or len(unit_hdr.coords) == 0:
+    measurement_system_code = segy_info.binary_header_dict[MEASUREMENT_SYSTEM_KEY]
+
+    if int(measurement_system_code) not in (1, 2):
+        logger.warning("Measurement system header is empty or corrupt. Can't extract coordinate unit.")
```

Comment:
Scary message!
```suggestion
        logger.warning("Unexpected value found for coordinate unit. Please double check your segy_spec for correct %s byte location.", MEASUREMENT_SYSTEM_KEY )
```

Maybe this should suggest `segy_spec` and `segy_header_overrides`?